itest/utils.go

```diff
@@ -148,6 +152,73 @@ func MineBlocks(t *testing.T, client *rpcclient.Client,
 	return blocks
 }
 
+type UTXORequest struct {
+	Type   lnrpc.AddressType
+	Amount int64
+}
+
+// SetAnchorUTXOs sets the wallet state for the sending node to a set of UTXOs
+// of a specific type and value.
+func SetNodeUTXOs(t *harnessTest, wallet *node.HarnessNode,
+	feeRate btcutil.Amount, reqs []*UTXORequest) {
+
+	minerAddr := t.lndHarness.Miner.NewMinerAddress()
+
+	// Drain any funds held by the node.
+	wallet.RPC.SendCoins(&lnrpc.SendCoinsRequest{
+		Addr:    minerAddr.EncodeAddress(),
+		SendAll: true,
+	})
+	t.lndHarness.MineBlocksAndAssertNumTxes(1, 1)
+
+	// Build TXOs from the UTXO requests, which will be used by the miner
+	// to build a TX.
+	makeOutputs := func(req *UTXORequest) *wire.TxOut {
+		addrResp := wallet.RPC.NewAddress(
+			&lnrpc.NewAddressRequest{
+				Type: req.Type,
+			},
+		)
+
+		addr, err := btcutil.DecodeAddress(
+			addrResp.Address, t.lndHarness.Miner.ActiveNet,
+		)
+		require.NoError(t.t, err)
+
+		addrScript, err := txscript.PayToAddrScript(addr)
+		require.NoError(t.t, err)
+
+		return &wire.TxOut{
+			PkScript: addrScript,
+			Value:    req.Amount,
+		}
+	}
+
+	aliceOutputs := fn.Map(reqs, makeOutputs)
+
+	_ = t.lndHarness.Miner.SendOutputsWithoutChange(aliceOutputs, feeRate)
+	t.lndHarness.MineBlocksAndAssertNumTxes(6, 1)
```

Comment:
Do we need to mine 6 blocks? 1 should be enough for most RPCs that require a min conf of 1.